nilearn/datasets/struct.py

```diff
@@ -139,22 +146,64 @@ def load_mni152_template():
        Human Brain Mapping 2009 Annual Meeting, DOI: 10.1016/S1053-8119(09)70884-5
 
     """
-    return check_niimg(MNI152_FILE_PATH)
 
+    img = check_niimg(MNI152_FILE_PATH)
+    brain_template = new_img_like(img, get_data(img).astype("float32"))
```

Comment:
why is this necessary?

Reply:
This is the typecasting in order to assure that we are using only np.float32.